File: Go/reverse_integer/7.go
package reverseinteger

import "math"

func reverse(x int) int {
	result := 0
	for x != 0 {
		last_number := x % 10
		result = result*10 + last_number
		x /= 10
	}
	if result > math.MaxInt32 || result < math.MinInt32 {
		return 0
	}
	return result
}

File: TypeScript/reverse_integer/7.ts
function reverse(x: number): number {
  const MAX = 2147483647;
  let result = 0;
  const isNegative = x < 0;

  if (isNegative) {
    x = x * -1;
  }

  while (x !== 0) {
    const lastNumber = x % 10;
    result = result * 10 + lastNumber;
    x = Math.floor(x / 10);
  }

  if (result > MAX) {
    return 0;
  }

  return isNegative ? -result : result;
}

export default reverse;
